### ISSUE TITLE
multiline strings

### ISSUE THREAD

**Issue by phase** [andrewrk/zig#76]
This is painful.
```
const vertex_shader_text: []u8 =
"uniform mat4 MVP;\n" ++
"attribute vec3 vCol;\n" ++
"attribute vec2 vPos;\n" ++
"varying vec3 color;\n" ++
"void main()\n" ++
"{\n" ++
"    gl_Position = MVP * vec4(vPos, 0.0, 1.0);\n" ++
"    color = vCol;\n" ++
"}\n";
```

I propose something more Pythonic, like
```
const vertex_shader_text: []u8 =
"""uniform mat4 MVP;
attribute vec3 vCol;
attribute vec2 vPos;
varying vec3 color;
void main()
{
    gl_Position = MVP * vec4(vPos, 0.0, 1.0);
    color = vCol;
}""";
```

Or just have any string be a multiline string. It wouldn't hurt existing code and it looks nice.
```
const vertex_shader_text: []u8 =
"uniform mat4 MVP;
attribute vec3 vCol;
attribute vec2 vPos;
varying vec3 color;
void main()
{
    gl_Position = MVP * vec4(vPos, 0.0, 1.0);
    color = vCol;
}";
```

**Comment by andrewrk** [comment]
Regardless of multiline string features, the `++` operator still has a purpose. It fills the role of `"foo" "bar"` in C - compile time string concatenation. It's useful when you have something like this:

```zig
const about_string = @compile_var("app_name") ++ "-" ++ @compile_var("version");
```

In this snippet, the compilation variables are available at compile time - they are the equivalent of C preprocessor defines.

Having any string be a multiline string is a reasonable idea, however, here are my reasons for not choosing it:

 * It might have been a typo to not put the end quote on the same line. If newline is invalid in normal string literals, the error message can be a lot better.
 * You might want quotes in your multiline string, and it would be a pain to escape them all, and it seems common you might want a quote in your multiline string.

Pythonic syntax looks very nice and seems tempting at first. But then, what if you want Python code in your multiline string? The point of a multiline string is to avoid having to escape anything. And I think we can accomplish that point by choosing the multiline string syntax carefully.

The C++11 syntax looks like this:

```
    add_simple_case("function pointers", R"SOURCE(
import "std.zig";

pub fn main(args: [][]u8) i32 => {
    const fns = []@typeof(fn1) { fn1, fn2, fn3, fn4, };
    for (f, fns) {
        print_u64(f());
        print_str("\n");
    }
    return 0;
}

fn fn1() u32 => {5}
fn fn2() u32 => {6}
fn fn3() u32 => {7}
fn fn4() u32 => {8}
    )SOURCE", "5\n6\n7\n8\n");
```

This is an example test case from run_tests.cpp. Here, "SOURCE" is an arbitrary string that begins and ends the multiline string. Because the user can use any string, no matter what you want to put in your multiline string, you can be guaranteed to not have to escape anything.

Another alternative is the Rust synatx, where you do: `r######"foo"######` - here you can put as many pound signs as you want to. This accomplishes the same goal: instead of escaping anything in the multiline string, just add more '#'.

**Comment by andrewrk** [comment]
I choose C++11 style multiline string literal syntax and implemented it in e144ddab249af3737f04267c7f1f0f0e093ed314.
